ls1intum/Artemis
# `Quiz exercises`: Fix reset question button of drag and drop questions in re-evaluation page

### Checklist
#### General


#### Client

### Motivation and Context



The question reset button of the DnD question on the quiz reevaluation page does not work. This caused the instructor not to be able to revert the changes made during quiz reevaluation.

https://user-images.githubusercontent.com/10885503/213447528-a1ad1cc3-73a3-491a-b807-c589bcc345d7.mov

### Description


The `questionEditorText` attribute of `DragAndDropQuestionEditComponent` needs to be updated with the new value of the markdown editor, otherwise, `questionEditorText` will always contain the same value (the first value when the component is loaded) which causes the reset functionality to be not working.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Quiz Exercise that is ended ([https://artemis-test1.artemis.in.tum.de/course-management/40/quiz-exercises/1916/re-evaluate](https://artemis-test1.artemis.in.tum.de/course-management/40/quiz-exercises/1916/re-evaluate))

1. Log in to Artemis
2. Navigate to Course Administration
3. Select the Course
4. Navigate to Exercises
5. Create a new Quiz Exercise
6. Set the duration to 1 Second
7. Create 1 Drag n Drop Question in the Quiz Exercise
8. Click Save and then Back
9. Click Start Exercise
10. Wait for 6 seconds, then refresh the page
11. Click the Reevaluation button of the quiz exercise
12. Update the question of DnD question
13. Click the undo button
14. Verify that the undo button works

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Branch | Line |
|------------|-------:|-----:|
| drag-and-drop-question-edit.component.ts | 77.27% | 88.85% |

### Screenshots



https://user-images.githubusercontent.com/10885503/213447553-f75845d8-ce64-4dfd-b270-9ed4d0b80f64.mov
